docs/dev/HOCInterpreter/HOCInterpreter.md

```diff
@@ -377,116 +379,138 @@ So given the following HOC call:
+      pnum = 0x00006000002101c0
+      pvalint = 0x00006000002101c0
+      pvalfloat = 0x00006000002101c0
+      u_auto = 2163136
+      ptr = 0x00006000002101c0
+      u_proc = 0x00006000002101c0 // this is the active member
+      rng = (type = 448, index = 24576)
+      ppsym = 0x00006000002101c0
+      ctemplate = 0x00006000002101c0
+      sym = 0x00006000002101c0
     }
+    s_varn = 0
+    arayinfo = nullptr
+    extra = nullptr
+    next = 0x0000600001709300
+  }
     
-    (lldb) p *fp->sp
-    (Symbol) $3 = {
-      name = 0x0000000100d07e00 "printf"
-      type = 280 // this is the type of the symbol, which corresponds to a builtin function: FUN_BLTIN = 280
-      subtype = 0
-      public = 0
-      defined_on_the_fly = 0
-      u = {
-        oboff = 13663760
-        pval = 0x0000000100d07e10
-        object_ = 0x0000000100d07e10
-        cstr = 0x0000000100d07e10 "??'"
-        pnum = 0x0000000100d07e10
-        pvalint = 0x0000000100d07e10
-        pvalfloat = 0x0000000100d07e10
-        u_auto = 13663760
-        ptr = 0x0000000100d07e10 (0x0000000100d07e10)
-        u_proc = 0x0000000100d07e10
-        rng = (type = 32272, index = 1)
-        ppsym = 0x0000000100d07e10
-        template = 0x0000000100d07e10
-        sym = 0x0000000100d07e10
-      }
-      s_varn = 0
-      arayinfo = 0x0000000000000000
-      extra = 0x0000000000000000
-      next = 0x0000000100d07e30
-    }
-    
-    (lldb) p *fp->argn
-    (Datum) $4 = {
-      val = 3 // value of the last argument (1+2) has already been computed
-      sym = 0x4008000000000000
-      i = 0
-      pval = 0x4008000000000000
-      pobj = 0x4008000000000000
-      obj = 0x4008000000000000
-      pstr = 0x4008000000000000
-      itm = 0x4008000000000000
-      lst = 0x4008000000000000
-      _pvoid = 0x4008000000000000
-    }
+  (lldb) p *fp->argn
+  (StackDatum) $3 =  Active Type = double  {
+    Value = 3 // value of the last argument (1+2) has already been computed
+  }
   ```  
 
 _**call()**_ will continue on to check the symbol, which  tells us we are calling a builtin function (_**FUN\_BLTIN = 280**_, see _**fp**_ above), and call its function pointer mapped to **_hoc\_PRintf_**: 
 
-   ```cpp   
-    if (sp->type == FUN_BLTIN || sp->type == OBJECTFUNC || sp->type == STRINGFUNC) {
-       stackp += sp->u.u_proc->nauto * 2; /* Offset stack for auto space */
-       STACKCHK
-       (*(sp->u.u_proc->defn.pf))(); /* this is where we call the printf function */
-       if (hoc_errno_check()) {
+  ```cpp
+  if (sp->type == FUN_BLTIN || sp->type == OBJECTFUNC || sp->type == STRINGFUNC) {
+      // Push slots for auto
+      for (auto i = 0; i < sp->u.u_proc->nauto; ++i) {
+          push_value(0.0);
+      }
+      (*(sp->u.u_proc->defn.pf))(); /* this is where we call the printf function */
+      if (hoc_errno_check()) {
           hoc_warning("errno set during call of", sp->name);
-       }
-    } else
-   ``` 
-
-![](images/58791243.png)
```

Comment:
The old image was a screenshot of lldb, I replaced it with pasted text.

Reply:
Hmm. I'm not seeing the sanitizer error on my M4 with the test0.hoc above.
```9.0.dev-97-g3f5d00d2a olupton/stack-datum```

Reply:
If I apply
```diff
diff --git a/src/oc/code.cpp b/src/oc/code.cpp
index 151ae198..66261f0b 100644
--- a/src/oc/code.cpp
+++ b/src/oc/code.cpp
@@ -1552,7 +1552,7 @@ void hocobjret(void) /* return from a hoc level obfunc */
     ret();
     /*make a temp and ref it in case autoobj returned since ret would
     have unreffed it*/
-    hoc_push_object(o);
+    hoc_push_object(*d);

     if (o) {
         o->refcount--;
```
(i.e. partially reverting your fix) then I see the error running `./bin/test/testneuron`

Reply:
> Hmm. I'm not seeing the sanitizer error on my M4 with the test1.hoc above.

It would only resurface in case of future underlying changes.